### PR TITLE
feature:  add label ai.tensorchord.envd.build.manifestBytecodeHash to image for cache robust

### PR DESCRIPTION
**Pull request by aseaday**
More accurate image cache based on the compiled bytecode'hash.
See e2e/bytecode_hash_test.go for more details.
https://github.com/tensorchord/envd/issues/631
https://github.com/tensorchord/envd/issues/553
Signed-off-by: Qi Chen <aseaday@hotmail.com>

### CODE REVIEW

File: pkg/types/label.go [hunk 1]
```diff
@@ -20,14 +20,15 @@ const (
...
-	ImageLabelContext = "ai.tensorchord.envd.build.context"
+	ImageLabelVendor    = "ai.tensorchord.envd.vendor"
+	ImageLabelGPU       = "ai.tensorchord.envd.gpu"
+	ImageLabelAPT       = "ai.tensorchord.envd.apt.packages"
+	ImageLabelPyPI      = "ai.tensorchord.envd.pypi.packages"
+	ImageLabelR         = "ai.tensorchord.envd.r.packages"
+	ImageLabelCUDA      = "ai.tensorchord.envd.gpu.cuda"
+	ImageLabelCUDNN     = "ai.tensorchord.envd.gpu.cudnn"
+	ImageLabelContext   = "ai.tensorchord.envd.build.context"
+	ImageLabelCacheHash = "ai.tensorchord.envd.build.manifestBytecodeHash"
```
Comment (gaocegege) [inline pkg/types/label.go:31]: Should we use ai.tensorchord.envd.build.hash?

**Review by gaocegege — COMMENTED** [review]
Request another review from @VoVAllen @kemingy 

LGTM

File: pkg/types/label.go (hunk 1 above)
Comment (aseaday) [reply pkg/types/label.go:31]: I think build.hash is no a good idea. But I really need a short term. The hash digested from the bytecode of `build.envd`. 
@VoVAllen @kemingy

Comment (VoVAllen) [reply pkg/types/label.go:31]: How about `manifestHash`? What does byte code mean here?

Comment (aseaday) [reply pkg/types/label.go:31]: It is the hash of manifest's bytecode not manifest directly.

Comment (kemingy) [reply pkg/types/label.go:31]: * short one: `ai.tensorchord.envd.build.digest`
* including the implementation: `ai.tensorchord.envd.build.bytecodeHash`

Comment (aseaday) [reply pkg/types/label.go:31]: the first LGTM, thx!

File: pkg/lang/frontend/starlark/interpreter.go [hunk 2]
```diff
@@ -53,6 +57,30 @@ func NewInterpreter(buildContextDir string) Interpreter {
 	}
 }
 
+func GetEnvdProgramHash(filename string) (uint64, error) {
```
Comment (kemingy) [inline pkg/lang/frontend/starlark/interpreter.go:60]: Will it be better to return the `strconv.FormatUint(hashsum, 16)` directly?

**Review by kemingy — APPROVED** [review]
LGTM

**Review by VoVAllen — APPROVED** [review]
LGTM! Thanks for your contribution!

**Comment by gaocegege** [comment]
```
------------------------------
• [FAILED] [1.043 seconds]
bytecode hash cache target
/home/runner/work/envd/envd/go/src/github.com/tensorchord/envd/e2e/bytecode_hash_test.go:40
  [It] add some blank to build.envd
  /home/runner/work/envd/envd/go/src/github.com/tensorchord/envd/e2e/bytecode_hash_test.go:42

  Unexpected error:
      <*withstack.withStack | 0xc0004ee360>: {
          cause: <*errutil.leafError | 0xc00082e520>{
              msg: "image ‹quick-start:e2etest› not found",
          },
          stack: [0x1199d80, 0x167b988, 0x163d4e8, 0x49eee1],
      }
      image quick-start:e2etest not found
  occurred
  In [It] at: /home/runner/work/envd/envd/go/src/github.com/tensorchord/envd/e2e/bytecode_hash_test.go:47
```

I will give it a retry.

**Comment by aseaday** [comment]
@gaocegege Fix it

**Review by gaocegege — APPROVED** [review]
LGTM :+1:
/lgtm

**Pull request merged**